sunpy/map/mapbase.py

```diff
@@ -476,12 +476,23 @@ def wcs(self):
         if w2.wcs.ctype[1].lower() in ("solar-y", "solar_y"):
             w2.wcs.ctype[1] = 'HPLT-TAN'
 
-        # GenericMap.coordinate_frame is implemented using this method, so we
-        # need to do this only based on .meta.
-        ctypes = {c[:4] for c in w2.wcs.ctype}
-        # Check that the ctypes contains one of these three pairs of axes.
-        if ({'HPLN', 'HPLT'} <= ctypes or {'SOLX', 'SOLY'} <= ctypes or {'CRLN', 'CRLT'} <= ctypes):
-            w2.heliographic_observer = self.observer_coordinate
+        # Set observer coordinate information
+        #
+        # Clear all the aux information that was set earlier. This is to avoid
+        # issues with maps that store multiple observer coordinate keywords.
+        # Note that we have to create a new WCS as it's not possible to modify
+        # wcs.wcs.aux in place.
+        header = w2.to_header()
+        for kw in ['crln_obs', 'dsun_obs', 'hgln_obs', 'hglt_obs']:
```

Comment:
you are missing `rsun_ref`: https://github.com/astropy/astropy/blob/master/astropy/wcs/src/wcslib_auxprm_wrap.c#L178-L182

Reply:
That's deliberate, since it's fine to set `rsun_ref` using the first call to `WCS` above; only the observer coordinate keywords need to be cleared here to avoid duplicate or misleading information that doesn't come from `m.observer_coordinate`.